compute_tools/src/monitor.rs

```diff
@@ -36,14 +36,16 @@ fn watch_compute_activity(compute: &ComputeNode) {
                     continue;
                 }
 
-                // Get all running client backends except ourself, use RFC3339 DateTime format.
+                // Get all running client backends except ourselves, use RFC3339 DateTime format.
+                let _ = cli.query("SET application_name = 'compute_monitor';", &[]);
                 let backends = cli
                     .query(
                         "SELECT state, to_char(state_change, 'YYYY-MM-DD\"T\"HH24:MI:SS.US\"Z\"') AS state_change
                          FROM pg_stat_activity
                          WHERE backend_type = 'client backend'
                             AND pid != pg_backend_pid()
-                            AND usename != 'cloud_admin';", // XXX: find a better way to filter other monitors?
+                            AND usename != 'cloud_admin'
+                            AND application_name != 'compute_monitor';",
```

Comment:
Therefore I left all other filters that you had. But also added new one for this specific app

Reply:
> Therefore I left all other filters that you had. But also added new one for this specific app

As I mentioned, if I connect from external app with application name `compute_monitor`, with your changes these backends won't be counted, isn't it?

Reply:
But the name compute_monitor can be easily changed to some random generated constant value if it’s the main point. What’s about the idea? Does it worth it? If so, I will change the app name

Reply:
Sorry, I don't really follow what's the problem we are trying to solve here. Filtering by user name worked perfectly well so far, I've left this XXX just in case we have any problems with it later, but we didn't

Reply:
Let me remove this comment and rollback changes about monitoring. Will leave only how do search for latest timestamp